src/PrestaShopBundle/EventListener/DemoModeEnabledListener.php

```diff
@@ -115,7 +117,11 @@ public function onKernelController(FilterControllerEvent $event)
             return;
         }
 
-        $this->showNotificationMessage($demoRestricted);
+        $isXmlHttpRequest = $event->getRequest()->isXmlHttpRequest();
+
+        if (!$isXmlHttpRequest) {
+            $this->showNotificationMessage($demoRestricted);
```

Comment:
I suggest you to make an early return. If it's ajax request, then go ahead and return `JsonResponse` here and continue with normal flow. 

Or at least move `$this->showNotificationMessage($demoRestricted);` into `else` block in line 138 to avoid this `if`.